### DISCUSSION TITLE
Onnx export with batch_size 1

### DISCUSSION THREAD

**Discussion by marisancans** [facebookresearch/detectron2#4516] [Q&A]
Hi, I am able to use your scripts to export model to onnx, however, input shape is 3, 800, 1202 and I would like it to be 1, 3, 800, 1202, I dont need dynamic batch sizes because my scripts later down the like breaks (I need to convert to TensorRT and use it in triton). How can I achieve this? I have spent couple of days trying to do it with graph surgeon and similar tools without success (converting to tensorRT breaks in triton, wierd errors). Thanks

Im using model: detectron2/configs/COCO-Detection/faster_rcnn_R_101_C4_3x.yaml

**Comment by Jain-Archit** [comment]
Hi @marisancans , 

I am also trying to convert faster-rcnn_R_50_fpn_3x.yaml however, I am facing errors while trying to convert the pytorch model to onnx using the export model script provided. 
The code for conversion I am using:


```
cfg = setup_cfg(config_path)
torch_model = build_model(cfg)
DetectionCheckpointer(torch_model).resume_or_load(weight_path)
torch_model.eval()
model = torch_model

def get_sample_inputs():
        original_image = detection_utils.read_image(image_path, format='BGR')
        # Do same preprocessing as DefaultPredictor
        aug = T.ResizeShortestEdge(
            [960, 960], 960
        )
        height, width = original_image.shape[:2]
        image = aug.get_transform(original_image).apply_image(original_image)
        image = torch.as_tensor(image.astype("float32").transpose(2, 0, 1))

        inputs = {"image": image, "height": height, "width": width}
        # Sample ready
        sample_inputs = [inputs]
        return sample_inputs

sample_inputs = get_sample_inputs()

torch.onnx.export(model, 
                  sample_inputs, 
                  "faster_rcnn.onnx",
                  do_constant_folding=True, opset_version=11, verbose=True,
                  export_params=True)
```

        
Could you please help me out in figuring what I am doing wrong here? Getting the following error:

> RuntimeError: Only tuples, lists and Variables supported as JIT inputs/outputs. Dictionaries and strings are also accepted but their usage is not recommended. But got unsupported type Instances
